### ISSUE TITLE
`parentmodule(Base) == Main`

### ISSUE THREAD

**Issue by ericphanson** [JuliaLang/julia#53574]
This seems weird to me:
```julia
julia> parentmodule(Base)
Main
```
Is it a bug or semantically correct?

**Comment by ericphanson** [comment]
```julia
julia> Base.is_root_module(Base)
true
```
Hm, I think it might be that I don't understand "root modules". Looking at the proposed new definition in https://github.com/JuliaLang/julia/pull/53778, it really seems like Base is the only odd one out though (a root module is a module that is it's own parent OR is Base).

**Comment by JeffBezanson** [comment]
Yes, this is not that big a deal but long story short it's because a lot of the package loading code is itself in Base so Base doesn't get loaded like a normal package. We could maybe fix it by moving the magic `__toplevel__` module to Core and changing how we include Base.jl.
